8.9.0 Release notes (#3536)
elastic/security-docs: docs/release-notes/8.9.asciidoc

> [[release-notes-8.9.0]]

Author: ```suggestion
[[release-notes-header-8.9.0]]
```

== 8.9.0

[discrete]
[[release-notes-8.9.0]]

> == 8.9.0

Author: ```suggestion
== 8.9
```

[discrete]
[[deprecations-8.9.0]]
==== Deprecations
* Removes the option to use the legacy navigation menu ({pull}158094[#158094]).

[discrete]
[[features-8.9.0]]
==== New Features
* Allows you to install the Cloud Security Posture Management (CSPM) integration via CloudFormation ({pull}159994[#159994]).
* Creates a new dashboard, Cloud Native Vulnerability Management, that provides an overview of vulnerabilities on your cloud hosts ({pull}159699[#159699]).
* Allows you to group vulnerabilities by resource (host) on the Vulnerabilities Findings page, and creates a Resource flyout that displays detailed vulnerability findings for individual hosts ({pull}159873[#159873], {pull}158987[#158987]).
* Adds a new custom dashboard, "Detection rule monitoring" ({pull}159875[#159875]).
* Allows you to anonymize event field values sent to AI Assistant ({pull}159857[#159857]).
* Adds a *Chat* button that opens AI Assistant to the alert details flyout ({pull}159633[#159633]).
* Updates AI Assistant to let you create and delete custom system prompts and default conversations ({pull}159365[#159365]).
* Adds a tour that introduces the EQL missing events feature ({pull}158863[#158863]).

> * Lets you add tags to alerts ({pull}157786[#157786]).

Author: Slight modification because we're using the term "alert tags" to differentiate alert tags from tags that you can add to cases and rules.

```suggestion
* Allows you to add alert tags ({pull}157786[#157786]).
```

> * Adds a tooltip to snoozed rules showing exactly when alerting will resume ({pull}157407[#157407]).

Reviewer: I think this is a usability enhancement rather than a new feature.

> * Adds a tooltip to snoozed rules showing exactly when alerting will resume ({pull}157407[#157407]).

Author: Resolved in [b31ba03](https://github.com/elastic/security-docs/pull/3536/commits/b31ba03e1c58232e337085042173fd513a2afcca).

[discrete]
[[enhancements-8.9.0]]
==== Enhancements
* Adds a *Last response* dropdown menu to the Rules table that allows you to filter rules by the status of their last execution ("Succeeded", "Warning", or "Failed") ({pull}159865[#159865]).
* Creates a Lens dashboard for monitoring the use of tokens by AI Assistant ({pull}159075[#159075]).
* Creates a connector for D3 Security ({pull}158569[#158569]).
* Improves the interface for adding and updating Elastic prebuilt rules ({pull}158450[#158450]).

> * Shows a rule's actions on its details flyout ({pull}158189[#158189]).

Reviewer: It's a whole page

```suggestion
* Shows a rule's actions on its details page ({pull}158189[#158189]).
```

> * Shows a rule's actions on its details flyout ({pull}158189[#158189]).

Reviewer 2: Am I correct here? Or is there a rule's details flyout I'm missing? 👀 
```suggestion
* Shows a rule's actions on its corresponding alerts details flyout ({pull}158189[#158189]).
```

Author: You're right, @jmikell821! @banderror actually noticed the [same thing](https://github.com/elastic/security-docs/pull/3536#discussion_r1269735006) and @benironside added the change. Thanks, all!

* Allows you to add Lens visualizations to cases from the visualization's *More actions* menu ({pull}154918[#154918]).

[discrete]
[[bug-fixes-8.9.0]]
==== Bug fixes
* Fixes a bug that restricted the number of cloud accounts which could appear on the Cloud Security Posture dashboard to 10 ({pull}157233[#157233]).

> * Fixes a bug that allowed you to save a rule that has an alert filter missing a query ({pull}159690[#159690]).

Reviewer 2: ```suggestion
* Fixes a bug that allowed you to save a rule with an alert filter missing a query ({pull}159690[#159690]).
```

> * Fixes unexpected filtering behavior on the Alert page. When you select a filter that excludes all alerts, an empty table now appears ({pull}160374[#160374]).

Reviewer 3: Currently, it is not very clear, what is the intention of this fix. Please feel free to suggest more changes to below suggestion. 

```suggestion
* Fixes unexpected filtering behavior on the Alert page. In previous version when you selected a filter value say `open` for the alert `Status` and there are no open alerts available, system would ignore `open` option and will show rest of the alerts even if they are `closed`. New behavior does not ignore `open` option and will show an empty table if there are not `open` alerts available ({pull}160374[#160374]).
```

Author: Hey, @logeekal! Appreciate the review. We generally aim to keep release notes summaries as succint as possible and then include a PR link so users can learn more if they're interested. Since your suggested revision highlights the before and after behavior, we're thinking about modifying our summary to something like:

* Fixes unexpected filtering behavior on the Alert page. Now, when you select a filter that excludes all alerts, an empty table now appears as expected ({pull}160374[#160374]).

Let us know what you think - thank you!

Reviewer 3: @nastasha-solomon , thanks for your response. That makes sense.  It is good know that PR will linked as well for the customers.

> * Fixes unexpected filtering behavior on the Alert page. When you select a filter that excludes all alerts, an empty table now appears ({pull}160374[#160374]).

Reviewer 2: ```suggestion
* Fixes unexpected filtering behavior on the Alerts page. Now, when you select a filter that excludes all alerts, an empty table appears ({pull}160374[#160374]).
```

Author: Resolved in [f3e26f8](https://github.com/elastic/security-docs/pull/3536/commits/f3e26f81e4ba16b68427369817365421e7cdc480).

* Improves input validation for investigation guide queries ({pull}160574[#160574], {pull}160577[#160577]).

> * Improves the handling of special characters in case tags ({pull}159815[#159815]).

Reviewer 2: Are these really bugs -- or enhancements?

Author: Thanks for the extra pair of eyes on this, @jmikell821! My responses below:
- #160574:  Looks like the PR fixes a UI bug where the **Label** field in the Investigation Guide form incorrectly turns red when the entered value is correct. Do you think we should modify the release notes summary so that it's closer to this description? 
- #159815:  This is a ResponseOps PR that should be doc'd in the Kibana release notes, so I'm removing it.

Reviewer 2: @nastasha-solomon re: #160574 - we can leave it as a bug, but let's start it with `Fixes`. I'm a +1 for revising it to the description you noted.

Author: Resolved in [397d33a](https://github.com/elastic/security-docs/pull/3536/commits/397d33a9dfa887ce7bb85fea5ef5df2c9a04b1c1).

> * Fixes a bug that could cause rules to snooze longer than intended ({pull}152873[#152873]).

Reviewer 2: ```suggestion
* Fixes a bug that caused rules to snooze longer than specified ({pull}152873[#152873]).
```

There are no breaking changes in 8.9.0.

//end::breaking-changes[]

[discrete]
[[deprecations-8.9.0]]
==== Deprecations
* Removes the option to use the legacy navigation menu ({pull}158094[#158094]).

[discrete]
[[features-8.9.0]]
==== New Features
* Allows you to install the Cloud Security Posture Management (CSPM) integration via CloudFormation ({pull}159994[#159994]).
* Creates a new dashboard, Cloud Native Vulnerability Management, that provides an overview of vulnerabilities on your cloud hosts ({pull}159699[#159699]).
* Allows you to group vulnerabilities by resource (host) on the Vulnerabilities Findings page, and creates a Resource flyout that displays detailed vulnerability findings for individual hosts ({pull}159873[#159873], {pull}158987[#158987]).
* Adds a new custom dashboard, "Detection rule monitoring" ({pull}159875[#159875]).
* Allows you to anonymize event field values sent to AI Assistant ({pull}159857[#159857]).
* Adds a *Chat* button that opens AI Assistant to the alert details flyout ({pull}159633[#159633]).
* Updates AI Assistant to let you create and delete custom system prompts and default conversations ({pull}159365[#159365]).

> * Adds a tour that introduces the EQL missing events feature ({pull}158863[#158863]).

Reviewer 4: Hey @nastasha-solomon, could we please remove this one? The tour has been removed because we are not releasing that feature in 8.9.0 anymore.

Author: ```suggestion
```

* Lets you add tags to alerts ({pull}157786[#157786]).
* Makes the Lateral Movement Detection advanced analytics package General Availability, and adds the ability to detect malicious activities in Windows RDP events (https://github.com/elastic/integrations/pull/6588[#6588])

[discrete]
[[enhancements-8.9.0]]

> ==== Enhancements

Reviewer 4: Could we please add an item for this [enhancement](https://github.com/elastic/kibana/pull/159029)? We made it easier to add exceptions by auto-populating alert highlighted fields and values.

Author: @peluja1012 sure, np! To clarify, do you want it to be doc'd as a new enhancement or a new feature? The PR's release note label indicates it's a new feature, so I just want to double check.

```suggestion
==== Enhancements
* Makes it easier to set up exceptions by auto-populating exception conditions and values with relevant alert data.  
```

* Adds a *Last response* dropdown menu to the Rules table that allows you to filter rules by the status of their last execution ("Succeeded", "Warning", or "Failed") ({pull}159865[#159865]).
* Creates a Lens dashboard for monitoring the use of tokens by AI Assistant ({pull}159075[#159075]).
* Creates a connector for D3 Security ({pull}158569[#158569]).
* Improves the interface for installing and upgrading Elastic prebuilt rules ({pull}158450[#158450]).
* Shows a rule's actions on its details page ({pull}158189[#158189]).
* Allows you to add Lens visualizations to cases from the visualization's *More actions* menu ({pull}154918[#154918]).
* Adds a tooltip to snoozed rules showing exactly when alerting will resume ({pull}157407[#157407]).

[discrete]
[[bug-fixes-8.9.0]]

> ==== Bug fixes

Reviewer 4: We added one more bug fix in 8.9: https://github.com/elastic/kibana/pull/159908

Author: Thanks! I'll add this in. 

Btw, I think our release notes tool missed https://github.com/elastic/kibana/pull/161793 as well because our script hadn't been updated to include the new `Team:Detection Engine` team label. I've added that team label to our script so it should pick up all DE PRs that are flagged for release notes now.

```suggestion
==== Bug fixes
* Fixes a bug that prevented rule exceptions from being auto-populated when you created a new exception from an alert's **Take action** menu.
* Fixes a UI bug that overlaid **Default Risk score** values as you created a new rule.
```

[[release-notes-header-8.9.0]]
== 8.9.0

[discrete]
[[release-notes-8.9.0]]
=== 8.9

[discrete]
[[known-issue-8.9.0]]
==== Known issues

* On the new Detection rule monitoring dashboard, total `Rule executions` will not always equal the sum of `Succeeded`, `Warning`, and `Failed` executions. This is expected because rules can write multiple statuses per execution. One typical example is gap detection: if a rule detects a gap in rule execution it will write an intermediate `Failed` status, then continue to run, and write a final status (such as `Warning`) before finishing its execution.

> * Rule changes can't be saved if the rule's action frequency is shorter than the rule's run interval.

Author: @yctercero please review this summary when you can. Thank you!

Reviewer 5: I like that - it's super succinct and to the point without exposing all the internals that would just confuse a user (and confuses us!) more.

> * The upload response action does not report the correct amount of available disk space. The correct amount is approximately four gigabytes. 

Author: @nfritts please review this summary when you can. Thank you!

[discrete]
[[breaking-changes-8.9.0]]
==== Breaking changes
//tag::breaking-changes[]
// NOTE: The breaking-changes tagged regions are reused in the Elastic Installation and Upgrade Guide. The pull attribute is defined within this snippet so it properly resolves in the output.
// THIS ALSO MEANS IF YOU USE LINKS HERE, THEY SHOULD BE FULL URLS WITH NO ATTRIBUTES

:pull: https://github.com/elastic/kibana/pull/

There are no breaking changes in 8.9.0.

//end::breaking-changes[]

[discrete]
[[deprecations-8.9.0]]
==== Deprecations
* Removes the option to use the legacy navigation menu ({pull}158094[#158094]).

> * Several prebuilt threat indicator match rules were deprecated and replaced with improved indicator type rules. 

Author: @w0rk3r please review this summary when you can. Thank you!

[discrete]
[[features-8.9.0]]
==== New features
* Allows you to install the Cloud Security Posture Management (CSPM) integration via CloudFormation ({pull}159994[#159994]).
* Creates a new dashboard, Cloud Native Vulnerability Management, that provides an overview of vulnerabilities on your cloud hosts ({pull}159699[#159699]).
* Allows you to group vulnerabilities by resource (host) on the Vulnerabilities Findings page, and creates a Resource flyout that displays detailed vulnerability findings for individual hosts ({pull}159873[#159873], {pull}158987[#158987]).
* Adds a new custom dashboard, "Detection rule monitoring" ({pull}159875[#159875]).
* Allows you to anonymize event field values sent to AI Assistant ({pull}159857[#159857]).
* Adds a *Chat* button that opens AI Assistant to the alert details flyout ({pull}159633[#159633]).
* Updates AI Assistant to let you create and delete custom system prompts and default conversations ({pull}159365[#159365]).
* Allows you to add alert tags ({pull}157786[#157786]).
* Adds the ability to automatically isolate a host through a rule’s endpoint response action. 
* Moves response actions to General Availability.

> * Adds a new response action that allows you to upload files to an endpoint that has {elastic-endpoint} installed.

Author: @kevinlog @caitlinbetz would you mind reviewing lines 46-48 when you have a moment? I couldn't find the PRs for these features so please let me know what those are if they should be included. Thank you!

Reviewer 6: I found a couple of possible PRs to reference, but not for all three lines.

[[release-notes-header-8.9.0]]
== 8.9.0

[discrete]
[[release-notes-8.9.0]]
=== 8.9

[discrete]
[[known-issue-8.9.0]]
==== Known issues

* On the new Detection rule monitoring dashboard, total `Rule executions` will not always equal the sum of `Succeeded`, `Warning`, and `Failed` executions. This is expected because rules can write multiple statuses per execution. One typical example is gap detection: if a rule detects a gap in rule execution it will write an intermediate `Failed` status, then continue to run, and write a final status (such as `Warning`) before finishing its execution.
* Rule changes can't be saved if the rule's action frequency is shorter than the rule's run interval.

> * The upload response action does not report the correct amount of available disk space. The correct amount is approximately four gigabytes. 

Reviewer 6: ```suggestion
* The `upload` response action does not report the correct amount of available disk space. The correct amount is approximately four gigabytes. 
```

[discrete]
[[breaking-changes-8.9.0]]
==== Breaking changes
//tag::breaking-changes[]
// NOTE: The breaking-changes tagged regions are reused in the Elastic Installation and Upgrade Guide. The pull attribute is defined within this snippet so it properly resolves in the output.
// THIS ALSO MEANS IF YOU USE LINKS HERE, THEY SHOULD BE FULL URLS WITH NO ATTRIBUTES

:pull: https://github.com/elastic/kibana/pull/

There are no breaking changes in 8.9.0.

//end::breaking-changes[]

[discrete]
[[deprecations-8.9.0]]
==== Deprecations
* Removes the option to use the legacy navigation menu ({pull}158094[#158094]).

> * Several prebuilt threat indicator match rules were deprecated and replaced with improved indicator type rules. 

Reviewer 7: ```suggestion
* General prebuilt threat indicator match rules were deprecated and replaced with improved indicator-type rules. 
```

[discrete]
[[features-8.9.0]]
==== New features
* Allows you to install the Cloud Security Posture Management (CSPM) integration via CloudFormation ({pull}159994[#159994]).
* Creates a new dashboard, Cloud Native Vulnerability Management, that provides an overview of vulnerabilities on your cloud hosts ({pull}159699[#159699]).
* Allows you to group vulnerabilities by resource (host) on the Vulnerabilities Findings page, and creates a Resource flyout that displays detailed vulnerability findings for individual hosts ({pull}159873[#159873], {pull}158987[#158987]).
* Adds a new custom dashboard, "Detection rule monitoring" ({pull}159875[#159875]).
* Allows you to anonymize event field values sent to AI Assistant ({pull}159857[#159857]).
* Adds a *Chat* button that opens AI Assistant to the alert details flyout ({pull}159633[#159633]).
* Updates AI Assistant to let you create and delete custom system prompts and default conversations ({pull}159365[#159365]).
* Allows you to add alert tags ({pull}157786[#157786]).

> * Adds the ability to automatically isolate a host through a rule’s endpoint response action. 
> * Moves response actions to General Availability.
> * Adds a new response action that allows you to upload files to an endpoint that has {elastic-endpoint} installed.

Reviewer 6: ```suggestion
* Adds the ability to automatically isolate a host through a rule’s endpoint response action ({pull}152424[#152424]). 
* Moves response actions to General Availability.
* Adds a new response action that allows you to upload files to an endpoint that has {elastic-endpoint} installed ({pull}157208[#157208]).
```

* Creates a Lens dashboard for monitoring the use of tokens by AI Assistant ({pull}159075[#159075]).
* Creates a connector for D3 Security ({pull}158569[#158569]).
* Improves the interface for installing and upgrading Elastic prebuilt rules ({pull}158450[#158450]).
* Shows a rule's actions on its details page ({pull}158189[#158189]).
* Allows you to add Lens visualizations to cases from the visualization's *More actions* menu ({pull}154918[#154918]).
* Adds a tooltip to snoozed rules that shows exactly when alerting will resume ({pull}157407[#157407]).
* Enhances the Data Exfiltration Detection package by adding the ability to detect exfiltration anomalies through USB devices and Airdrop (https://github.com/elastic/integrations/pull/6577[#6577]).

[discrete]
[[bug-fixes-8.9.0]]
==== Bug fixes
* Fixes a bug that prevented rule exceptions from being auto-populated when you created a new exception from an alert's **Take action** menu.
* Fixes a UI bug that overlaid **Default Risk score** values as you created a new rule.

> * Fixes a bug that restricted the number of cloud accounts which could appear on the Cloud Security Posture dashboard to 10 ({pull}157233[#157233]).

Reviewer 6: ```suggestion
* Fixes a bug that restricted the number of cloud accounts that could appear on the Cloud Security Posture dashboard to 10 ({pull}157233[#157233]).
```
